lib/inspec/resources/oracle.rb

```diff
@@ -0,0 +1,52 @@
+require "inspec/resources/powershell"
+
+module Inspec::Resources
+  class Oracle < Inspec.resource(1)
+    name "oracle"
+    supports platform: "unix"
+    supports platform: "windows"
+
+    desc "The 'oracle' resource is a helper for the 'oracledb_listener_conf'"
+
+    attr_reader :conf_path
+
+    def initialize
+      case inspec.os[:family]
+      when "debian", "redhat", "linux", "suse"
+        determine_conf_dir_and_path_in_linux
+      when "windows"
+        determine_conf_dir_and_path_in_windows
+      end
+    end
+
+    def to_s
+      "OracleDB"
+    end
+
+    private
+
+    def determine_conf_dir_and_path_in_linux
+      oracle_home = inspec.command("echo $ORACLE_HOME").stdout&.chomp
```

Comment:
I'm confused - why not just read `ENV["ORACLE_HOME"]` here?

Reply:
This is the value which is set in the system inside bash file. This value is mostly set during Oracle installation itself so I was using that. Setting this value inside environment variable would an added task. Should we be making use do that?

Reply:
And I just observed I mentioned above it was an environment key so it was not the environment key. Just correcting that fact.

Reply:
I am little confused too now, Can you elaborate on this? 
Do you mean something like `inspec.command("ENV['ORACLE_HOME']")` or this `ENV['ORACLE_HOME']` ?

Reply:
There is no need to use `inspec.command` to read an environment variable - just read directly from ENV.

Reply:
So I gave it some thought and also tried what you suggested. Using `ENV['ORACLE_HOME']` is using the value from the local system ( where the InSpec is running ). But it could be pointed to some other target machine, like in my case one is linux and another is windows and they could have a dynamic path depending on the version of oracle and platforms.

So I feel it might be better to fetch value which is set in the target system through command.Thoughts?

Reply:
Very good point. In that case, we should use `inspec.os_env('ORACLE_HOME').content` to read the value instead of command - we have a resource just for reading env variables.

Reply:
okay I will give this a try. Thanks :)